Deploy more smart pointers in Source/WebKit/NetworkProcess/webrtc/

rniwa opened the pull request:
#### 20fa016440f2f3704f85bdab5669007b4c82fff7
<pre>
Deploy more smart pointers in Source/WebKit/NetworkProcess/webrtc/
<a href="https://bugs.webkit.org/show_bug.cgi?id=279577">https://bugs.webkit.org/show_bug.cgi?id=279577</a>

Reviewed by Youenn Fablet.

Deploy more smart pointers in these files per clang static analyzer warnings.

* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::NetworkRTCProvider):
(WebKit::NetworkRTCProvider::startListeningForIPC):
(WebKit::NetworkRTCProvider::close):
(WebKit::NetworkRTCProvider::createResolver):
(WebKit::NetworkRTCProvider::signalSocketIsClosed):
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.cpp:
(WebKit::RTCDataChannelRemoteManagerProxy::registerConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::unregisterConnectionToWebProcess):

Canonical link: <a href="https://commits.webkit.org/283570@main">https://commits.webkit.org/283570@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/52670606628f1b074458466240fcc227e0f68c39

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66605 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45980 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19227 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70639 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17738 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53779 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17500 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53380 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11961 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69672 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42332 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57637 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34046 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39003 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16092 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60914 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15369 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72341 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10562 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14732 "Found 1 new test failure: http/wpt/cache-storage/quota-third-party.https.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60703 "Found 2 new test failures: http/tests/security/referrer-policy-header-multipart.html http/tests/security/referrer-policy-header.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10594 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57708 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61032 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8689 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2311 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10115 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41787 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42864 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44047 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42607 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
```diff
@@ -118,6 +118,9 @@ class NetworkRTCProvider : private FunctionDispatcher, private IPC::MessageRecei
     IPC::Connection& connection() { return m_ipcConnection.get(); }
     Ref<IPC::Connection> protectedConnection() { return m_ipcConnection.get(); }
 
+    RefPtr<NetworkConnectionToWebProcess> protectedConnectionToWebProcess();
```
This getter does not seem to be used in this patch.

rniwa replied:
Oh weird. Will remove.

youennf commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
```diff
@@ -157,8 +160,6 @@ class NetworkRTCProvider : private FunctionDispatcher, private IPC::MessageRecei
 
     void assertIsRTCNetworkThread();
 
-    Ref<NetworkRTCMonitor> protectedRTCMonitor();
```
It does not seem protectedRTCMonitor is used elsewhere than in NetworkRTCProvider.

rniwa replied:
Oh, right. This just got removed. Will revert that.

rniwa committed:
Deploy more smart pointers in Source/WebKit/NetworkProcess/webrtc/
https://bugs.webkit.org/show_bug.cgi?id=279577

Reviewed by Youenn Fablet.

Deploy more smart pointers in these files per clang static analyzer warnings.

* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::NetworkRTCProvider):
(WebKit::NetworkRTCProvider::startListeningForIPC):
(WebKit::NetworkRTCProvider::close):
(WebKit::NetworkRTCProvider::createResolver):
(WebKit::NetworkRTCProvider::signalSocketIsClosed):
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.cpp:
(WebKit::RTCDataChannelRemoteManagerProxy::registerConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::unregisterConnectionToWebProcess):

Canonical link: https://commits.webkit.org/283570@main
